### DISCUSSION TITLE
官网上的两个调优参数计算结果不知道怎么算的，求大神解答

### DISCUSSION THREAD

**Discussion by lianneli** [apache/celeborn#1280] [Q&A]
在官网上看到两个参数计算方式。只有条件和结果，不知道怎么计算出来的。请恕我才疏学浅，求大神给出详细解答。
1. estimatedTasks是怎么算的？怎么能从10个numdirs和256KB buffersize算出来10G的必需offheap？
![image](https://user-images.githubusercontent.com/19421187/221512677-2cf666cd-527f-4672-9388-95310d90ff41.png)

2. 单record的buffer size是64KB的情况下，怎么算出来最大的inflight records是1310720？
![image](https://user-images.githubusercontent.com/19421187/221513163-ace3e066-106d-439f-825c-ec69669e54cb.png)

**Comment by FMX** [comment]
@lianneli  回复的有点晚了，你的问题解答如下：
1. ， 256KB 一个 flush buffer，双副本，10个目录，每个目录的 buffers 总数是 2048。（注：最新版本已经去掉了每个目录的 buffers 上限）
2.  5 个节点，每个节点 16 GB 共 80GB。inFlight  参数我们理解为是可以直接发送到网络上的数据块的数量，这里 80GB/64KB 就得到了 1310720。

我们有钉钉群和slack，可以在上面即时沟通。
文档中有部分内容已经过时了。我最近把文档更新掉。
